src/mixer/basetrackplayer.h

```diff
@@ -100,6 +107,10 @@ class BaseTrackPlayerImpl : public BaseTrackPlayer {
     ControlObjectSlave* m_pPreGain;
     ControlObjectSlave* m_pSpeed;
     ControlObjectSlave* m_pPitchAdjust;
+    QScopedPointer<ControlObjectSlave> m_pInputEnabled;
+    QScopedPointer<ControlObjectSlave> m_pPassthroughEnabled;
+    QScopedPointer<ControlObjectSlave> m_pVinylControlEnabled;
+    QScopedPointer<ControlObjectSlave> m_pVinylControlStatus;
```

Comment:
> Please remember, that I have compared a scoped pointer solutions with a plain delete. Both are redundant here because of the QObject tree which offers the same convenience. 

Indeed, and compared to the smart pointer bits we're talking about that costs much more since it involves removing the child from the parent's children list and broadcasting a "child removed" event. :)